src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php

```diff
@@ -61,18 +68,38 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $shouldForce = $input->getOption('force');
         $ids = (array) $input->getArgument('id');
-        $shouldDisplayMessages = $input->getOption('show-messages') || 1 === \count($ids);
-        $this->removeMessages($failureTransportName, $ids, $receiver, $io, $shouldForce, $shouldDisplayMessages);
+        $shouldDeleteAllMessages = $input->getOption('all');
 
-        return 0;
-    }
+        $idsCount = \count($ids);
+        if (!$shouldDeleteAllMessages && !$idsCount) {
+            throw new RuntimeException('Please specify at least one message id. If you want to remove all messages, use the "--all" option.');
+        } elseif ($shouldDeleteAllMessages && $idsCount) {
+            throw new RuntimeException('You cannot specify message ids when using the "--all" option.');
+        }
+
+        if ($shouldDeleteAllMessages && !$shouldForce) {
+            $io->error('Please provide the "--force" option when using "--all" to confirm the removal of all failed messages in the transport.');
```

Comment:
please check how we do when --force is not passed but and $id is: I don't think we display at the error level nor that we return 1, do we? also, we should display something useful, like the number or messages that would be deleted

Reply:
Alright, I misunderstood the comment above and took example on another command like `TranslationUpdateCommand` that also defines the `--force` option. I updated accordingly. Without `--force`, the number of messages that would removed is displayed as well as a hint that encourage to use `--force` with `--all`